src/liquid/components/ld-sidenav/ld-sidenav.tsx

```diff
@@ -189,6 +233,83 @@ export class LdSidenav {
     if (this.initialized) {
       this.collapsed = false
     }
+
+    setTimeout(() => {
+      this.updateFocus()
+    })
+  }
+
+  @Listen('keydown', { passive: true, target: 'window' })
+  handleKeyDown(ev: KeyboardEvent) {
+    // Ignore events if sidenav has no focus and is not closable.
+    const hasSidenavFocus =
+      document.activeElement.closest('ld-sidenav') === this.el
+    if (!hasSidenavFocus && !this.closable) {
+      return
+    }
+
+    const slider = this.el.querySelector('ld-sidenav-slider')
+
+    if (ev.key === 'Escape') {
+      if (!slider && this.closable) {
+        this.open = false
+        return
+      }
+
+      if (this.hasActiveSubnav) {
+        this.handleSlideBack()
+      } else if (this.closable) {
+        this.open = false
+      }
+    }
+  }
+
+  @Listen('focusout', { passive: true, target: 'window' })
+  handleFocusout(ev: FocusEvent) {
+    // If the sidenav is closable, trap the focus.
+    // Do not trap the focus as long as the sidenav is not closable or not open.
+    if (!this.closable || !this.open) return
+
+    // Do not trap the focus if the trap focus prop is not set.
+    if (typeof this.trapFocus === 'undefined') return
+
+    const relatedTarget = ev.relatedTarget as HTMLElement | undefined
+
+    // Do not trap the focus as long as the focus remains within the sidenav.
+    const isFocusInSidenav = relatedTarget?.closest('ld-sidenav') === this.el
+    if (isFocusInSidenav) return
+
+    // Do not trap the focus when it moves to an element which matches
+    // the trap focus selector.
+    const isFocusInKeepFocusable =
+      this.trapFocus !== '' && relatedTarget?.matches(this.trapFocus)
+    if (isFocusInKeepFocusable) return
+
+    // Keep the focus where it was before.
+    const target = ev.target as HTMLElement
+    const lastFocused = target.shadowRoot
+      ? getFirstFocusable(target.shadowRoot)
+      : target
+
+    lastFocused.focus()
+  }
+
+  private toFocus: HTMLElement = undefined
+  private focusTimeout = undefined
+  private updateFocus = () => {
+    // HACK: Using recursive call with timeout to account for fast user interactions during transitions.
```

Comment:
This seems to work quite well. But if you hit [Tab] during the transition to a subnav, the focus gets "lost" (as the parent nav is not yet fully hidden).